onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -397,17 +397,37 @@ void AttentionTypeAndShapeInference(ONNX_NAMESPACE::InferenceContext& ctx, int p
 
     auto& bias_shape = getInputShape(ctx, 2);
     auto& bias_dims = bias_shape.dim();
-    if (bias_dims.size() != 1 || bias_shape.dim(0).dim_value() % 3 != 0) {
+    if (bias_dims.size() != 1) {
       fail_shape_inference("Invalid bias shape");
     }
 
+    std::vector<int64_t> qkv_hidden_sizes;
+    getRepeatedAttribute(ctx, "qkv_hidden_sizes", qkv_hidden_sizes);
+
+    if (hasInputShape(ctx, extra_add_index)) {
```

Comment:
Do not need extra_add in this function since output shape/type are not deduced from extra_add

Reply:
I see, thanks for clarifying the purpose of this fn, updated.